frontend/src/e2e-playwright/pages/employee/vasu/vasu.ts

```diff
@@ -2,21 +2,136 @@
+  readonly #primaryPhoneNumberInput = this.findAll(
+    this.primaryInputSelector
+  ).nth(3)
+
+  readonly #primaryValue = this.findAll(
+    '[data-qa="multi-field-question"] [data-qa="value-or-no-record"]'
+  ).first()
+
+  readonly #otherFieldsRows = this.findAll(
+    '[data-qa="multi-field-list-question"] [data-qa="field-row"]'
+  )
+  readonly otherFieldsInputSelector = 'input'
+  readonly #otherFieldsValues = this.findAll(
+    '[data-qa="value-or-no-record"]'
+  ).nth(1)
+
+  get primaryFirstNameInput(): TextInput {
+    return new TextInput(this.#primaryFirstNameInput)
+  }
+
+  get primaryLastNameInput(): TextInput {
+    return new TextInput(this.#primaryLastNameInput)
+  }
+
+  get primaryTitleInput(): TextInput {
+    return new TextInput(this.#primaryTitleInput)
+  }
+
+  get primaryPhoneNumberInput(): TextInput {
+    return new TextInput(this.#primaryPhoneNumberInput)
+  }
+
+  get primaryValue(): Promise<string | null> {
+    return this.#primaryValue.textContent
+  }
+
+  get otherFieldsCount(): Promise<number> {
+    return this.#otherFieldsRows.count()
+  }
+
+  otherFirstNameInput(ix: number): TextInput {
+    return new TextInput(
+      this.#otherFieldsRows
+        .nth(ix)
+        .findAll(this.otherFieldsInputSelector)
+        .first()
+    )
+  }
+
+  otherLastNameInput(ix: number): TextInput {
+    return new TextInput(
+      this.#otherFieldsRows
+        .nth(ix)
+        .findAll(this.otherFieldsInputSelector)
+        .nth(1)
+    )
+  }
+
+  otherTitleInput(ix: number): TextInput {
+    return new TextInput(
+      this.#otherFieldsRows
+        .nth(ix)
+        .findAll(this.otherFieldsInputSelector)
+        .nth(2)
+    )
+  }
+
+  otherPhoneNumberInput(ix: number): TextInput {
+    return new TextInput(
+      this.#otherFieldsRows
+        .nth(ix)
+        .findAll(this.otherFieldsInputSelector)
+        .nth(3)
+    )
+  }
 
+  otherValues(): Promise<string | null> {
+    return this.#otherFieldsValues.textContent
+  }
+}
+
+class VasuPageCommon {
+  constructor(readonly page: Page) {}
+
+  readonly #documentSection = this.page.findAll(
+    '[data-qa="vasu-document-section"]'
+  )
   readonly #followupQuestions = this.page.findAll(
     '[data-qa="vasu-followup-question"]'
   )
-  readonly finalizeButton = this.page.find(
-    '[data-qa="transition-button-MOVED_TO_READY"]'
-  )
+
+  getDocumentSection(ix: number) {
+    // Note: indexes might change if the template used in the test changes
+    return this.#documentSection.nth(ix)
+  }
+
+  async assertDocumentVisible() {
+    await this.#documentSection.first().waitUntilVisible()
+  }
+
+  get authorsSection(): AuthorsSection {
+    return new AuthorsSection(this.getDocumentSection(0))
+  }
+
+  get followupQuestionCount(): Promise<number> {
+    return this.#followupQuestions.count()
+  }
+}
+
+export class VasuEditPage extends VasuPageCommon {
+  constructor(page: Page) {
+    super(page)
+  }
```

Comment:
Constructor should not be needed if it only calls the base class constructor with the same params? Or maybe it is 🤷